lib/pure/strutils.nim

```diff
@@ -325,7 +394,8 @@ proc toOctal*(c: char): string {.noSideEffect, rtl, extern: "nsuToOctal".} =
     result[i] = chr(val mod 8 + ord('0'))
     val = val div 8
 
-iterator split*(s: string, seps: set[char] = Whitespace, maxsplit: int = -1): string =
+iterator split*(s: string, seps: set[char] = Whitespace, maxsplit: int = -1,
+                keepSeparators: bool = false): string =
```

Comment:
``keepSeparators`` is overkill, ``split`` is not a tokenizer. Also 2 yield statements in an inline iterator cause code bloat (loop body will inserted twice).